### PR TITLE
Add backward compatibility loss functions for new error and strict imitation under Tensorflow implmentation together with example notebooks

### PR DESCRIPTION
**Pull request by ilmarinen**
Currently still troubleshooting Nans in one loss function.

### CODE REVIEW

**Review by nushib — APPROVED** [review]

**Review by Nking92 — APPROVED** [review]
This all looks great to me. The code is clean, well documented, and all the notebooks ran perfectly.

**Review by jlema — APPROVED** [review]
